bridges/snowbridge/primitives/core/src/fees.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+use log;
+use sp_runtime::{DispatchResult, SaturatedConversion, Saturating, TokenError};
+use xcm::opaque::latest::{Location, XcmContext};
+use xcm_executor::traits::TransactAsset;
+const LOG_TARGET: &str = "xcm_fees";
+
+/// Burns the fees embedded in the XCM for teleports.
+pub fn burn_fees<AssetTransactor, Balance>(dest: Location, fee: Balance) -> DispatchResult
```

Comment:
another note about it:

since it is defined in `primitives`, maybe make it less opinionated:

```suggestion
/// Burns the fees embedded in the XCM for teleports.
pub fn burn_assets_for_teleport<AssetTransactor>(dest: Location, assets: Assets) -> DispatchResult
```

this way you can use it for various types of asset(s)

Reply:
Renamed: https://github.com/vgeddes/polkadot-sdk/blob/snowbridge-v2/bridges/snowbridge/primitives/core/src/lib.rs#L191